vdev/src/commands/package/msi.rs

```diff
@@ -0,0 +1,149 @@
+        {
+            println!("Sorry, the package-msi command is not supported on non-Windows platforms. Nothing was performed.");
+        }
+        #[cfg(windows)]
+        {
+            let archive_version = app::version()?;
+
+            let package_version = {
+                let channel = util::release_channel()?;
+                if channel == "custom" {
+                    // Extract the version before ".custom"
+                    archive_version.split(".custom").next().unwrap().to_string()
+                } else {
+                    archive_version.clone()
+                }
+            };
+            // Make sure we start with a fresh `target/msi-x64` target directory and
+            // copy the `distribution/msi` directory to `target/msi-x64`
+            let msi_x64_dir = Path::new("target").join("msi-x64");
+            fs::remove_dir_all(&msi_x64_dir).ok();
+            fs::create_dir_all(&msi_x64_dir)?;
+            fs::copy("distribution/msi", &msi_x64_dir)?;
+
+
+            let artifacts_dir = Path::new("target").join("artifacts");
+            let zip_file = format!("vector-{archive_version}-x86_64-pc-windows-msvc.zip");
+            fs::copy(artifacts_dir.join(&zip_file), msi_x64_dir.join(&zip_file))?;
+
+            // Ensure we are in the `msi-x64` directory
+            env::set_current_dir(&msi_x64_dir)?;
+
+            // Extract the zip file with PowerShell and build the MSI package
+            let powershell_command = format!(
+                "$progressPreference = 'silentlyContinue'; Expand-Archive {zip_file}"
+            );
+            app::exec("powershell", ["-Command", &powershell_command], false)?;
+            build(&archive_version, &package_version)?;
+
+            // Change the current directory back to the original path
+            env::set_current_dir(app::path())?;
+
+            // Copy the MSI file to the artifacts directory
+            let msi_file = format!("vector-{archive_version}-x64.msi");
+            let dest_file = artifacts_dir.join(msi_file);
+            fs::copy(msi_x64_dir.join("vector.msi"), dest_file)?;
+        }
+        Ok(())
+    }
+}
+
+#[cfg(windows)]
+fn build(archive_version: &str, package_version: &str) -> Result<()> {
+    println!("Running Build with args: {archive_version}");
+    println!("Copying ZIP archive...");
+
+    println!("Preparing LICENSE.rtf..");
+    let mut license_rtf_file = File::create("LICENSE.rtf")?;
+    writeln!(
+        license_rtf_file,
+        "{{\\rtf1\\ansi\\ansicpg1252\\deff0\\nouicompat{{\\fonttbl{{\\f0\\fnil\\fcharset0 Lucida Console;}}}}\n\\viewkind4\\uc1\n\\pard\\f0\\fs14\\lang1033\\par"
+    )?;
+
+    let license_content_path = format!("vector-{archive_version}-x86_64-pc-windows-msvc/LICENSE.txt");
+    let license_content = std::fs::read_to_string(license_content_path)?;
+    for line in license_content.lines() {
+        writeln!(license_rtf_file, "{line}\\")?;
+    }
+    writeln!(license_rtf_file, "\n}}")?;
```

Comment:
I think the `\n` needs to be escaped here:
```suggestion
    writeln!(license_rtf_file, "\\n}}")?;
```

Reply:
I don't think this will be required since \\n will put backslash and then n when the objective was to create a newline.

I would like to have this tested though with a windows machine to run candle and light commands on the generated files to see if they are correct format. I need to test this out fully and verify the output of the generated files.

Reply:
Ah, I missed that it was an `echo -e` in the original script, which interprets those escapes. Not something I think I've ever actually used in a script.